test/expire_cache_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class ExpireCacheTest < IdentityCache::TestCase
+  def setup
+    super
+
+    Item.cache_index(:title, unique: true)
+    Item.cache_index(:id, :title, unique: true)
+
+    @record = Item.new
+    @record.id = 1
+    @record.title = "bob"
+
+    Spy.on(Item.cached_primary_index, :load_one_from_db).and_return do
+      @record
+    end
+  end
+
+  def test_expire_cache_hit
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+    assert(@record.expire_cache)
+    assert_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+  end
+
+  def test_expire_cache_on_failure
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+
+    Spy.on(backend, :write_entry).and_return { false }
+
+    refute(@record.expire_cache)
+    refute_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+  end
+end
```

Comment:
This is missing a test for `expire_cache` returning `false` when `expire_primary_index` fails to invalidate a cache key.

This is also missing tests for when embedded cache associations are expired.  That is done by `expire_parent_caches`, but it looks like that isn't called as part of `expire_cache`.  I suppose that would have been fine before when `expire_cache` was just an internal API, but now that it is public, we should probably move the call to `expire_parent_caches` from `_run_commit_callbacks` into `expire_cache` for correctness.

Reply:
I've moved expire parent caches to `expire_cache` method and include it in result as well